File: frontend/src/types/colony.ts
import { Cell } from './cell';
import { Directive } from './directive';
import { Resources } from './resources';

export interface Colony {
  id: string;
  owner_id: string;
  name: string;
  width: number;
  height: number;
  cells: Cell[][];
  created_at: Date;
  last_update: Date;
  tick_count: number;
  resources: Resources;
  active_directives: Directive[];
  auto_save: boolean;
}

File: backend/internal/models/colony.go
package models

import "time"

type Colony struct {
	ID          string      `json:"id"`
	OwnerID     string      `json:"owner_id"`
	Name        string      `json:"name"`
	Width       int         `json:"width"`
	Height      int         `json:"height"`
	Cells       [][]Cell    `json:"cells"`
	CreatedAt   time.Time   `json:"created_at"`
	LastUpdate  time.Time   `json:"last_update"`
	TickCount   int64       `json:"tick_count"`
	Resources   Resources   `json:"resources"`
	ActiveDirectives []Directive `json:"active_directives"`
	AutoSave       bool     `json:"auto_save"`
}

type ColonyMetadata struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	OwnerID       string    `json:"owner_id"`
	Width         int       `json:"width"`
	Height        int       `json:"height"`
	CellCount     int       `json:"cell_count"`
	LastUpdate    time.Time `json:"last_update"`
	ThumbnailData []byte    `json:"thumbnail_data"`
}